Add Go tab to the cookboook (#129)
amzn/ion-docs: guides/cookbook.md

If Ion binary encoding is desired, use `IonBinaryWriterBuilder` (instead of `IonTextWriterBuilder`).
</div>

<div class="tabpane Go" markdown="1">
Implementations of the [`Reader`][22] and [`Writer`][23] interfaces are responsible for reading and writing Ion data in both text and binary forms.

> In order to make and use a text reader, we can use `NewReaderString()`. The following example demonstrates how to read Ion data from a string:

Reviewer: ```suggestion
In order to create and then use a text reader, we can use `NewReaderString()`. The following example demonstrates how to read Ion data from a string:
```

	if reader.Next() {                                           // position the reader at the first value
		currentType := reader.Type()                             // the first value in the reader is a struct
		fmt.Println("Current type is:\t" + currentType.String()) // Current type is:   struct
		reader.StepIn()                                          // step into the struct
		reader.Next()                                            // position the reader at the first value in the struct
		currentType = reader.Type()                              // the first value in the struct is of type string
		fmt.Println("Current type, after stepping in the struct:\t" +
			currentType.String())          // Current type, after stepping in the struct:   string
		fieldName := reader.FieldName()    // retrieve the current value's field name
		value, err := reader.StringValue() // retrieve the current value's string value
		if err != nil {
			panic("Reading string value failed.")
		}
		reader.StepOut()                    // step out of the struct
		fmt.Println(*fieldName, " ", value) // hello world
	}
}
```
If we have a binary Ion, `NewReaderBytes()` can be used in the same fashion.

> To write the the same struct as above, `{hello:"world"}`, we can use `NewTextWriter()` as shown below:

Reviewer: ```suggestion
To write the  same struct as above, `{hello:"world"}`, we can use `NewTextWriter()` as shown below:
```

```
</div>

<div class="tabpane Go" markdown="1">

A reader can have various types of numeric values:
  - Int32
  - Int64
  - Uint64
  - BigInt
  - Float
  - Decimal

> The following example illustrates how to read different numeric values in a reader:

Reviewer: ```suggestion
The following example illustrates how to read different numeric values:
```

```Go
package main

import (
	"fmt"
	"math/big"

	"github.com/amzn/ion-go/ion"
)

func main() {
	int32Value := 2147483646

> 	var int64Value int64 = 9223372036854775807

Reviewer: ```suggestion
	 int64Value int64 := 9223372036854775807
```

Not sure why we need `var` only for this line and not the others

Reviewer 2: I needed to explicitly define a value as int64 so I can test ```reader.Int64Value()```.
Assignment like ```int64Value := 9223372036854775808```, will infer ```value64``` as type of ```Int```.

For consistency, changed the definition here and explicitly cast it to ```int64``` on line 1241.

```
</div>

<div class="tabpane Go" markdown="1">

```Go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
	
	"github.com/amzn/ion-go/ion"
)

func main() {

> 	file, er := os.Open("c://values.cvs")

Reviewer: ```suggestion
	file, er := os.Open("test.csv")
```
